support/Environments/sonic.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/env sh
+
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+spectre_setup_modules() {
+    echo "All modules on Sonic are provided by the system"
+}
+
+spectre_load_modules() {
+    # The order here is important
+    module load use.own
+    # module load gcc/11.1.0
+    module load spectre-env > /dev/null 2>&1
+}
+
+spectre_unload_modules() {
+    # The order here is important
+    module unload spectre-env > /dev/null 2>&1
+    # module unload gcc/11.1.0
+    module unload use.own
+}
+
+spectre_run_cmake_gcc_release() {
+    if [ -z ${SPECTRE_SOURCE} ]; then
+        echo "You must set SPECTRE_SOURCE to the cloned SpECTRE directory"
```

Comment:
Also to keep this consistent with other clusters, we require that `SPECTRE_HOME` be set

Reply:
I would really like to keep it consistent. However I noticed the usage of SPECTRE_ROOT in some places, and SPECTRE_HOME, making it confusing. I then decided to keep the environment variables self-explanatory, SPECTRE_SOURCE for the source code, SPECTRE_BUILD for the build directory, and SPECTRE_INSTALL for the install directory.  I also checked that this does not have any bearing on the running of SpECTRE itself, as it is not being used in anywhere in the source code, apart from some examples and documentation.

Please let me know if you have any concerns.

Reply:
Yes you are right that we aren't consistent with `SPECTRE_ROOT` vs `SPECTRE_HOME` throughout the code. We'll have to fix that eventually. And I do understand your reasoning for wanting to use different names. However, we *are* consistent with `SPECTRE_HOME` in all of our cluster environment scripts and I think we should keep the consistency. Otherwise, when adding new clusters, people will be confused why Sonic is different from the others. We want to make the experience of using spectre on different clusters as uniform as possible.

Reply:
> Yes you are right that we aren't consistent with `SPECTRE_ROOT` vs `SPECTRE_HOME` throughout the code. We'll have to fix that eventually. And I do understand your reasoning for wanting to use different names. However, we _are_ consistent with `SPECTRE_HOME` in all of our cluster environment scripts and I think we should keep the consistency. Otherwise, when adding new clusters, people will be confused why Sonic is different from the others. We want to make the experience of using spectre on different clusters as uniform as possible.

Just trying to adopt good build practices. If SpECTRE code guidelines mandate this, we will be happy to adhere to this.